dbms/src/TestUtils/tests/gtest_mock_executors.cpp

```diff
@@ -195,6 +236,32 @@ try
                           "      table_scan_0 | {<0, Long>, <1, String>, <2, String>}\n";
         ASSERT_DAGREQUEST_EQAUL(expected, request);
     }
+
+    {
+        auto right_builder = context.scan("test_db", "r_table")
+                                 .filter(And(eq(col("r_table.r_a"), col("r_table.r_b")), eq(col("r_table.r_a"), col("r_table.r_b"))))
+                                 .project({col("r_table.r_a"), col("r_table.r_b"), col("join_c")})
+                                 .aggregation({Max(col("r_table.r_a"))}, {col("join_c"), col("r_table.r_b")})
+                                 .topN({{"r_table.r_b", false}}, 10);
+
+        auto left_builder = context.scan("test_db", "l_table")
+                                .topN({{"l_table.l_a", false}}, 10)
+                                .join(right_builder, tipb::JoinType::TypeLeftOuterJoin, {col("join_c")}) // todo ensure the join is legal.
+                                .limit(10);
+        request = left_builder.build(context);
+        {
+            String expected = "limit_8 | 10\n"
+                              " Join_7 | LeftOuterJoin, HashJoin. left_join_keys: {<0, String>}, right_join_keys: {<0, String>}\n"
```

Comment:
why `<0, String>` or not `<3, String>`?

Reply:
> why `<0, String>` or not `<3, String>`?
Probably related to executorSerializer.cpp:50, toString function print the column from index 0.

Reply:
> why `<0, String>` or not `<3, String>`?

Related to executorSerializer.cpp:50, toString function print the column from index 0.